src/config.rs

```diff
@@ -17,21 +17,46 @@ pub fn get_config_path() -> String {
     }
 }
 
-/// Reads configuration from file and panics if it can't
+// Tries to read configuration from file, failing that from the environment,
+// panics if it can't
 pub fn read_config() -> Config {
+    match read_config_from_file() {
+        Ok(config) => config,
+        Err(err) => {
+            println!("{}", err);
+            match read_config_from_env() {
+                Ok(config) => config,
+                Err(err) => panic!("{}", err)
+            }
+        }
+    }
+}
+
+/// Tries to read configuration from file
+pub fn read_config_from_file() -> Result<Config, String> {
     let config_path = get_config_path();
 
-    let contents = fs::read_to_string(&config_path).unwrap_or_else(|_| {
-        panic!("Failed to open config file at {}", config_path);
-    });
-    let config: Config = toml::from_str(contents.as_str()).unwrap_or_else(|_| {
-        panic!("Failed to parse config file at {}", config_path);
-    });
+    let contents = fs::read_to_string(&config_path).map_err(|_| {
+        format!("Failed to open config file at {}", config_path)
+    })?;
+    let config: Config = toml::from_str(contents.as_str()).map_err(|_| {
+        format!("Failed to parse config file at {}", config_path)
+    })?;
+
+    println!("Reading config from file at {}", config_path);
```

Comment:
Should probably state that it has completed reading. Something like: `"Config read from file at {}"`.